packages/discovery-react-components/src/components/DiscoverySearch/DiscoverySearch.tsx

```diff
@@ -287,31 +289,24 @@ const DiscoverySearch: FC<DiscoverySearchProps> = ({
       backwardsCompatibleQueryParams: DiscoveryV2.QueryParams & { returnFields?: string[] },
       resetAggregations = true
     ): Promise<void> => {
-      let aggregationsFetched = false;
       const searchParameters = deprecateReturnFields(
         backwardsCompatibleQueryParams
       ) as DiscoveryV2.QueryParams;
       setSearchParameters(searchParameters);
-      // don't use the search response if filter is set, just do another search
+
+      // TODO proper fix is to have any caller of handleSearch that needs aggregations
+      // to call handleFetchAggregations instead
```

Comment:
Is the reason to not do this now to minimize the number of changes in this patch?

Reply:
yes